src/app_service/common/network_constants.nim

```diff
@@ -32,6 +32,24 @@ let INFURA_TOKEN_SECRET_RESOLVED* =
   else:
     INFURA_TOKEN_SECRET
 
+# set via `nim c` param `-d:ALCHEMY_ETHEREUM_MAINNET_TOKEN:[token]`; should be set in CI/release builds
+const ALCHEMY_ETHEREUM_MAINNET_TOKEN {.strdefine.} = ""
+let ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV = $getEnv("ALCHEMY_ETHEREUM_MAINNET_TOKEN")
+let ALCHEMY_ETHEREUM_MAINNET_TOKEN_RESOLVED* =
+  if ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV != "":
+    ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV
+  else:
+    ALCHEMY_ETHEREUM_MAINNET_TOKEN
```

Comment:
Do we really need to repeat the same thing like 8 or more times? Could we not make a function for this?

Reply:
can do

Reply:
done